### PR TITLE
drivers: pinctrl: pfc_rcar: add dummy function to pinctrl driver

### PR DESCRIPTION
**Pull request by xakep-amatop**
Add a dummy pin function to the RCar PFC driver. It is necessary to ensure that the driver sets the 'peripheral' bit (the driver resets this bit during the first call of 'pfc_rcar_set_gpsr') for a pin that doesn't have a pin function defined in IPSR, but always acts as a 'peripheral', for example, the MMC pins on the Spider board.

### CODE REVIEW

**Comment by xakep-amatop** [comment]
@lorc @gmarull @aaillet Could you please review this PR?

**Comment by gmarull** [comment]
lgtm, will aprove once authors are ok

File: include/zephyr/dt-bindings/pinctrl/renesas/pinctrl-rcar-common.h
Comment (lorc) [inline include/zephyr/dt-bindings/pinctrl/renesas/pinctrl-rcar-common.h:29]: I am not sure that this is the good idea in terms of future proofing. IPSR entry is 4 bit wide and theres is a chance that some Renesas SoC will use all 16 values, including 0xF. For example, Gen3 already uses functions in range 0x0-0xD.

File: include/zephyr/dt-bindings/pinctrl/renesas/pinctrl-r8a779f0.h
Comment (lorc) [inline include/zephyr/dt-bindings/pinctrl/renesas/pinctrl-r8a779f0.h:279]: Well, this is not technically true. As you said in the commit message, those pin functions are not defined in IPSR registers, so it would be incorrect to say that `MMC_SD_CLK` is configured with bit 0 in `IP0SR1`. 

(Yes, I saw that MMC_SD_CLK is described under IP0SR1 section in the HW manual, but I tend to see this as a poor documentation choice).

Also, please take a look at note below, regarding `FUNC_DUMMY` value.

So, I want to propose introduce "dummy" register instead of "dummy function". This will fix both issues: potential use of function 0xF in the next SoC generation and incorrect assignment of all static functions to bit 0 of `IP0SR1`.

**Comment by xakep-amatop** [comment]
I'll update this PR after merging [the linked one](https://github.com/zephyrproject-rtos/zephyr/pull/67357) to avoid conflicts.

File: include/zephyr/dt-bindings/pinctrl/renesas/pinctrl-rcar-common.h
Comment (lorc) [inline include/zephyr/dt-bindings/pinctrl/renesas/pinctrl-rcar-common.h:85]: Please add comment similar to `PIN_NOGPSR_START` /  `RCAR_NOGP_PIN`

Comment (xakep-amatop) [reply include/zephyr/dt-bindings/pinctrl/renesas/pinctrl-rcar-common.h:95]: done

**Review by lorc — APPROVED** [review]

**Comment by xakep-amatop** [comment]
@gmarull @aaillet could you please review this PR?

**Review by gmarull — APPROVED** [review]

**Pull request merged**
